pyaedt/generic/ibis_reader.py

```diff
@@ -0,0 +1,748 @@
+        self._enable = None
+
+    @property
+    def name(self):
+        """Name of the item."""
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    @property
+    def model_type(self):
+        """Type of the model."""
+        return self._model_type
+
+    @model_type.setter
+    def model_type(self, value):
+        self._model_type = value
+
+    @property
+    def clamp(self):
+        """Clamp."""
+        return self._clamp
+
+    @clamp.setter
+    def clamp(self, value):
+        self._clamp = value
+
+    @property
+    def enable(self):
+        """Is model enabled or not."""
+        return self._enable
+
+    @enable.setter
+    def enable(self, value):
+        self._enable = value
+
+
+class Ibis:
+    """Ibis model with all data extracted: name, components, models.
+
+    Parameters
+    ----------
+    name : str
+        Name of ibis model.
+    circuit : class:`pyaedt.circuit.Circuit`
+        Circuit in which the ibis components will be used.
+    """
+
+    # Ibis reader must work independently or in Circuit.
+    def __init__(self, name, circuit):
+        self.circuit = circuit
+        self._name = name
+        self._components = {}
+        self._model_selectors = []
+        self._models = []
+
+    @property
+    def name(self):
+        """Name of the ibis model."""
+        return self._name
+
+    @property
+    def components(self):
+        """List of all components included in the ibis file."""
+        return self._components
+
+    @components.setter
+    def components(self, value):
+        self._components = value
+
+    @property
+    def model_selectors(self):
+        """List of all model selectors included in the ibis file."""
+        return self._model_selectors
+
+    @model_selectors.setter
+    def model_selectors(self, value):
+        self._model_selectors = value
+
+    @property
+    def models(self):
+        """List of all models included in the ibis file."""
+        return self._models
+
+    @models.setter
+    def models(self, value):
+        self._models = value
+
+    @property
+    def buffers(self):
+        """Buffers included into the ibis model."""
+        return self._buffers
+
+    @buffers.setter
+    def buffers(self, value):
+        self._buffers = value
+
+
+class IbisReader:
+    def read_project(self, filename, circuit):
+        """Reads *.ibis file content.
+
+        Parameters
+        ----------
+        filename : str
+            Name of ibis model.
+        circuit : class:`pyaedt.circuit.Circuit`
+            Circuit in which the ibis components will be used.
+
+        Returns
+        ----------
+        :class:`pyaedt.generic.ibis_reader.Ibis`
+            Ibis object exposing all data from the ibis file.
+        """
+
+        if os.path.exists(filename) == False:
+            error_message = filename + " does not exist."
+            raise FileExistsError(error_message)
```

Comment:
There’s no reason to assign and use a single use variable (unless for readability). Also, `FileExistsError` would be the wrong exception in Python 3, you’d use `FileNotFoundError`, however to maintain compatibility with IronPython 2.7, we would either have to create a custom exception or just use `OSError` or just a plain `Exception`.